common/src/massWindow.cc

```diff
@@ -0,0 +1,184 @@
+    
+    double interpolation = m_matrix[n][m].Interpolate(point.first, point.second);
+
+    double a_closestEllipse = 0;
+    double b_closestEllipse = 0;
+    double theta_closestEllipse = 0;
+    double dist = 0;
+    if (interpolation == 0) {
+        dist = 1000000000;
+        std::ifstream ifile(m_filename);
+        Json::Reader reader;
+        Json::Value text;
+        if (ifile && reader.parse(ifile, text)) {
+            for (int i=0; i<text.size(); i++) {
+                const double mbb = text[i][0].asDouble();
+                const double mllbb = text[i][1].asDouble();
+                const double a = text[i][2].asDouble();
+                const double b = text[i][3].asDouble();
+                const double theta = text[i][4].asDouble();
+                const double MA = text[i][5].asDouble();
+                const double MH = text[i][6].asDouble();
+                if (MA > 0 && MH > 0 && theta > 0 && a > 0 && b > 0) {
+                    double distance = sqrt(pow(point.first-mbb, 2)+pow(point.second-mllbb, 2));
+                    if (distance < dist){
+                        dist = distance;
+                        a_closestEllipse = a;
+                        b_closestEllipse = b;
+                        theta_closestEllipse = theta;
+                    }
+                }
+                // If no min dist found:
+                if (dist == 1000000000) interpolation = 0.;
+                else {
+                    if (n==0 && m==0) interpolation = cos(theta_closestEllipse)/sqrt(a_closestEllipse);
+                    else if (n==0 && m==1) interpolation = sin(theta_closestEllipse)/sqrt(a_closestEllipse);
+                    else if (n==1 && m==0) interpolation = -sin(theta_closestEllipse)/sqrt(b_closestEllipse);
+                    else if (n==1 && m==1) interpolation = cos(theta_closestEllipse)/sqrt(b_closestEllipse);
+                }
+            }
+        }
+    }
+    return interpolation;
+}
+
+
+pair_d massWindow::applyGlobalTransformation(pair_d ref, pair_d point) {
+
+    double m1 = this->getValue(0,0,ref)*point.first + this->getValue(0,1,ref)*point.second;
+    double m2 = this->getValue(1,0,ref)*point.first + this->getValue(1,1,ref)*point.second;
+    return std::make_pair(m1, m2);
+}
+
+
+pair_d massWindow::applyLocalTranformation(pair_d point) {
+
+    return this->applyGlobalTransformation(point,point);   
+}
+
+
+bool massWindow::isInWindow(double center_x, double center_y, double size, double point_x, double point_y){
+    
+    double m1diff = point_x - center_x;
+    double m2diff = point_y - center_y;
+    pair_d mdiff = std::make_pair(m1diff, m2diff);
+    pair_d center = std::make_pair(center_x, center_y);
+    pair_d param = this->applyGlobalTransformation(center, mdiff);
+    return sqrt(pow(param.first,2) + pow(param.second,2)) < size;
+}
+
+
+bool massWindow::isNoise(double center_x, double center_y, double size, double point_x, double point_y) {
```

Comment:
I would personally prefer some more explicit naming:
`isInWindow` -> `isInEllipse`
`isNoise` -> `isOutOfEllipse` 
if you wouldn't mind ?